framework/generated/vulkan_generators/vulkan_constant_maps_generator.py

```diff
@@ -0,0 +1,117 @@
+        protect_file=False,
+        protect_feature=True,
+        extraVulkanHeaders=[]
+    ):
+        BaseGeneratorOptions.__init__(
+            self,
+            blacklists,
+            platform_types,
+            filename,
+            directory,
+            prefix_text,
+            protect_file,
+            protect_feature,
+            extraVulkanHeaders=extraVulkanHeaders
+        )
+
+
+class VulkanConstantMapsGenerator(BaseGenerator):
+    """VulkanColorspaceMapGenerator - subclass of BaseGenerator.
+    """
+
+    def __init__(
+        self, err_file=sys.stderr, warn_file=sys.stderr, diag_file=sys.stdout
+    ):
+        BaseGenerator.__init__(
+            self,
+            process_cmds=True,
+            process_structs=False,
+            feature_break=True,
+            err_file=err_file,
+            warn_file=warn_file,
+            diag_file=diag_file
+        )
+
+    def beginFile(self, gen_opts):
+        """Method override."""
+        BaseGenerator.beginFile(self, gen_opts)
+
+        write('#include "util/defines.h"', file=self.outFile)
+        self.newline()
+        self.includeVulkanHeaders(gen_opts)
+        self.newline()
+        write('#include <unordered_map>', file=self.outFile)
+        write('#include <vector>', file=self.outFile)
+
+        write('GFXRECON_BEGIN_NAMESPACE(gfxrecon)', file=self.outFile)
+        write('GFXRECON_BEGIN_NAMESPACE(decode)', file=self.outFile)
+        self.newline()
+
+        colorspace_extensions_names = set()
+        colorspace_extension_map = dict.fromkeys(
+            filter(
+                lambda enum_value: 'VK_COLOR_SPACE' in enum_value,
+                self.registry.enumdict
+            )
+        )
+        colorspace_extension_map.pop(
+            'VK_COLOR_SPACE_SRGB_NONLINEAR_KHR'
+        )  # Remove the core provided color space from extension map
+        for extension in self.registry.extensions:
+            colorspace_enum = filter(
+                lambda element: 'VK_COLOR_SPACE' in element.get('name'),
+                extension.findall('require/enum')
+            )
+            colorspace_enum = map(
+                lambda element: element.get('name'), colorspace_enum
+            )
+            for name in colorspace_enum:
+                value = extension.get("name")
+                colorspace_extensions_names.add(value)
+                colorspace_extension_map[name] = f'"{value}"'
+
+        write(
+            'static const std::unordered_map<VkColorSpaceKHR, const char *> kColorSpaceExtensionMap = {',
+            file=self.outFile
+        )
+        for key, value in colorspace_extension_map.items():
+            write(f'{{{key}, {value}}},', file=self.outFile)
+        write('};', file=self.outFile)
+        self.newline()
+
+        write(
+            'static const std::vector<const char *> kColorSpaceExtensionNames = {',
+            file=self.outFile
+        )
+        for extension_name in colorspace_extensions_names:
```

Comment:
```suggestion
        for extension_name in sorted(colorspace_extensions_names):
```
This should be enough to fix the lack of ordering in the generator output. Turns out set() is unordered, while dict() maintains insertion order in python 3.7+. But since we only need the order during printing, we can sort it before output like so.

Reply:
Went ahead and pushed the fix. Its a one liner after all.